sympy/combinatorics/fp_groups.py

```diff
@@ -360,6 +361,38 @@ def index(self, H, strategy="relator_based"):
             C = self.coset_enumeration(H, strategy)
             return len(C.table)
 
+    def is_cyclic(self):
+        """
+        Return ``True`` if the group is Cyclic.
+
+        Examples
+        ========
+
+        >>> from sympy.combinatorics.free_groups import free_group
+        >>> from sympy.combinatorics.fp_groups import FpGroup
+        >>> F, x, y = free_group("x, y")
+        >>> f = FpGroup(F, [x*y*x**-1*y**-1, y**5, x**4])
+        >>> f.is_cyclic()
+        True
+        >>> f = FpGroup(F, [x**4, y**2, x*y*x**-1*y])
+        >>> f.is_cyclic()
+        False
+
+        """
+        if len(self.generators) == 1:
+            return True
+        if not self.is_abelian:
+            return False
+        for p in primefactors(self.order()):
+            pgens = []
+            for g in self.generators:
+                pgens.append(g**p)
+            K, T = self.subgroup(pgens, homomorphism=True)
```

Comment:
Actually both the `is_cyclic` methods are not completely same there is a slight difference so, I doubt it if a single method would be suffice or not.

Reply:
Do you mean that `is_cyclic` has different meanings for permutation groups and finitely presented groups?

Reply:
Actually we are using the method `.subgroup` which has been defined separately for both the permutation groups and fp groups also we are keeping `homomorphism=True` in the subgroup method so that we can have a list of generators otherwise it will simply produce the free variables and we will not be able to compute the `index` below.

But this is not the case in permutation group where the subgroup method directly returns the generators of the group via which index can be calculated easily.

Reply:
I guess `is_cyclic` is same for both but it actually depends on the implementation of subgroup method which generate the generators differently.

Reply:
Would it be possible to apply `PermutationGroup.is_cyclic` also for an instance of `FpGroup` by transforming it to a permutation group using `_to_perm_group`? One could first check the 0-1 generator cases.

Reply:
Yes, I think we can do this.
```
>>> F, x, y = free_group("x, y")
>>> f = FpGroup(F, [x*y*x**-1*y**-1, y**5, x**4])
>>> g, t = f._to_perm_group()
>>> g.is_cyclic()
True
>>> g
PermutationGroup([
    (0 1 5 2)(3 6 12 8)(4 7 13 9)(10 14 18 16)(11 15 19 17),
    (0 3 10 11 4)(1 6 14 15 7)(2 8 16 17 9)(5 12 18 19 13)])

```